pkg/workloads/deployment.go

```diff
@@ -101,3 +105,41 @@ func RequestDeployment(request DeploymentRequest) (*appsv1.Deployment, error) {
 
 	return deployment, nil
 }
+
+func CreateDeploymentWithSuffix(suffix string, component string, cr *argoproj.ArgoCD) (*appsv1.Deployment, error) {
+	lbls := cr.Labels
+	lbls[common.AppK8sKeyComponent] = component
+	deploymentRequest := DeploymentRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.NameWithSuffix(cr.Name, suffix),
+			Namespace: cr.Namespace,
+			Labels:    lbls,
+		},
+	}
+
+	deployment, err := RequestDeployment(deploymentRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return deployment, nil
+}
+
+// TriggerDeploymentRollout will update the label with the given key to trigger a new rollout of the Deployment.
+func TriggerDeploymentRollout(client cntrlClient.Client, deployment *appsv1.Deployment, key string) error {
+	currentDeployment, err := GetDeployment(deployment.Name, deployment.Namespace, client)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	currentDeployment.Spec.Template.ObjectMeta.Labels[key] = NowNano()
+	return UpdateDeployment(currentDeployment, client)
+}
```

Comment:
@ciiay 

I'm thinking a better approach for this might be to define this method that only takes in a deployment name, and some modification function, runs that modification and then updates the deployment. Something like
```
func TriggerDeploymentRollout( deployment *appsv1.Deployment, o option) error {
	currentDeployment, err := GetDeployment(deployment.Name, deployment.Namespace, client)
	if err != nil {
		if !errors.IsNotFound(err) {
			return err
		}
		return nil
	}
        // run options fn
	o()
	updateDeployment()
	return
}
```
and each controller that needs this should define this for themselves, so that the final call on these deployments is made by the controller they belong to